src/algebra/big_operators/fin.lean

```diff
@@ -139,6 +142,40 @@ by simpa only [prod_univ_add, fintype.prod_eq_one _ hf, mul_one]
 
 end fin
 
+/-- Equivalence between `fin n → fin m` and `fin (m ^ n)`. -/
```

Comment:
Can you add a lemma of the form `fin_function_fin_equiv (pi.single i j) = _` or similar (maybe with a coe)?